libs/stats/odc/stats/_gm_bitmask.py

```diff
@@ -0,0 +1,146 @@
+"""
+Landsat QA Pixel Geomedian
+"""
+from functools import partial
+from typing import Optional, Tuple
+
+import dask.array as da
+import xarray as xr
+from odc.algo import geomedian_with_mads, keep_good_only
+from odc.algo._masking import _xr_fuse, _first_valid_np, _fuse_and_np
+from odc.algo.io import load_with_native_transform
+from odc.stats.model import Task
+
+from . import _plugins
+from .model import StatsPluginInterface
+
+
+class StatsGMLSBitMask(StatsPluginInterface):
+    NAME = "gm_ls_annual"
+    SHORT_NAME = NAME
+    VERSION = "3.0.0"
+    PRODUCT_FAMILY = "geomedian"
+
+    def __init__(
+            self,
+            bands: Optional[Tuple[str, ...]] = None,
+            mask_band: str = "QA_PIXEL",
+            aux_names=dict(smad="sdev", emad="edev", bcmad="bcdev", count="count"),
+            rgb_bands=None,
+            resampling: str = "bilinear",
+            work_chunks: Tuple[int, int] = (400, 400),
+            **other,
+    ):
+        self.mask_band = mask_band
+        self.resampling = resampling
+        self.bands = bands
+        self.work_chunks = work_chunks
+        self.renames = aux_names
+        self.aux_bands = list(aux_names.values())
+
+        if bands is None:
+            bands = (
+                "red",
+                "green",
+                "blue",
+                "nir",
+                "swir1",
+                "swir2",
+            )
+            if rgb_bands is None:
+                rgb_bands = ("red", "green", "blue")
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        return self.bands + self.aux_bands
+
+    def _native_tr(self, xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads pq bands
+        2. Extract cloud_mask flags from bands
+        3. Add cloud_mask
+        4. Drops nodata pixels
+
+        .. bitmask::
+            7   6   5   4   3   2   1   0
+            |   |   |   |   |   |   |   |
+            |   |   |   |   |   |   |   x-----> nodata
+            |   |   |   |   |   |   o---------> dilated_cloud
+            |   |   |   |   |   x-------------> cirrus
+            |   |   |   |   o-----------------> cloud
+            |   |   |   x---------------------> cloud_shadow
+            |   |   o-------------------------> snow
+            |   x-----------------------------> clear
+            o---------------------------------> water
+        """
+        mask_band = xx[self.mask_band]
+        xx = xx.drop_vars([self.mask_band])
+
+        # set cloud_mask (cloud + cloud_shadow) bitmask
+        cloud_mask = da.bitwise_and(mask_band, 0b00011000) == 0
```

Comment:
I think the `QA_PIXEL` is 16 bit (I missed this earlier - sorry!) so this needs to have eight more zeros:

```suggestion
        cloud_mask = da.bitwise_and(mask_band, 0b0000_0000_0001_1000) == 0
```